src/Symfony/Component/Translation/Translator.php

```diff
@@ -438,4 +471,17 @@ private function getConfigCacheFactory()
 
         return $this->configCacheFactory;
     }
+
+    private function getFormatter($name)
+    {
+        $name = $name ?: $this->defaultFormatter;
+
+        foreach ($this->formatters as $formatter) {
+            if ($name === $formatter->getName()) {
```

Comment:
Can't you store the name as array key when registering?
This loop will be executed for **every** trans() call!